src/main_server.cpp

```diff
@@ -818,25 +830,22 @@ void handle_client_speaker(bool* connectionShutdownNotice,
 
                 if (success == true) {
                     //
-                    packet ackPkt;
-                    ackPkt.type = ACK;
-                    ackPkt.seqn = 0;
-                    ackPkt.length = 0;
-                    ackPkt.timestamp = curPkt.timestamp;
-                    strcpy(ackPkt._payload, curPkt._payload);
+                    Message ackPkt;
+                    ackPkt.set_type(Type::ACK);
+                    ackPkt.set_timestamp(curPkt.get_timestamp());
+                    ackPkt.set_payload(curPkt.get_payload());
```

Comment:
we should create a constructor here in the Message to receive payload, then inside we set the timestamp, as it won't work (getter doesn't create a timestamp, only returns the member)